lib/vanagon/project.rb

```diff
@@ -82,6 +82,32 @@ def get_files
       files.flatten.uniq
     end
 
+    # Returns a filtered out set of components only including those
+    # components necessary to build a specific component. This is a
+    # recursive function that will call itself until it gets to a
+    # component with no build requirements
+    #
+    # @param name [String] name of component to add. must be present in configdir/components and named $name.rb currently
+    def filter_component(name)
+      components_to_include = []
+      filter_component = get_component(name)
+      return nil if filter_component.nil?
+      components_to_include << filter_component
+      filter_component.build_requires.each do |build_requirement|
+        unless get_component(build_requirement).nil?
+          components_to_include += filter_component(build_requirement)
+        end
+      end unless filter_component.build_requires.nil?
+      components_to_include
+    end
```

Comment:
There is a LOT going on in this method name, and one of your variable names is also the same as your method name. This is taking me a little longer to unpack than it should, so that's probably a code smell.